### ISSUE TITLE
Resize window doesn't work on Wayland

### ISSUE THREAD

**Issue by cucumber-sp** [electron/electron#44543]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

33.0.2

### What operating system(s) are you using?

Other Linux

### Operating System Version

Arch Linux, Gnome 47.1

### What arch are you using?

x64

### Expected Behavior

![Image](https://github.com/user-attachments/assets/50377ab5-322b-42c9-88d1-4671a4916c8e)
Works when running with x

### Actual Behavior

![Image](https://github.com/user-attachments/assets/ba9d12e5-44a0-43f4-b7f8-a604dceb6942)
Ignores pointer when running with wayland

**Comment by cbrnr** [comment]
I can confirm, and as I've said in https://github.com/electron/electron/issues/42894#issuecomment-2449452459, this issue was introduced in Electron 32 (so the last working version was 31).

**Comment by EmptyVisual** [comment]
You can reproduce this issue by downloading the latest GitHub release (electron-v33.1.0-linux-x64 at the time of testing). To see the issue, run the binary as `./electron --ozone-platform-hint=auto` on a Wayland session. 
The problem will become apparent as you observe the inability to resize the client window.

The last known working version that allowed window resizing was v31, although it had the CSD border issue described in #42894  It seems likely that a change introduced between v31 and the current release may be causing this behavior.

Please let me know if any additional details or specific configurations are needed.

**Comment by TheNexter** [comment]
Same issue using webcord on wayland :
https://github.com/SpacingBat3/WebCord/issues/576

**Comment by mhoffmeyerDC** [comment]
Did a little bit of testing the various releases to try to narrow down the source of this regression. Found that it was not present in `v32.0.0-alpha.4` but is present in `v32.0.0-alpha.5` 

It's caused by #42681 and #41769 ... @codebytere even says in the PR:

> Functionality on Wayland will be approached as a follow-up as I don't currently have the requisite hardware to test this.

**Comment by machinaeXlasse** [comment]
Not sure if this is of any help but I figured that, after I use key combo to move the window to the left or right half of the screen ( [super/windows] + [left/right])  the resize option reappears and I can resize the window.

I observed this for VS Code and Signal Messenger.

I'm on Ubuntu 24.04 64bit Gnome 46 with Wayland.

**Comment by cbrnr** [comment]
Is there any chance this bug can be escalated in severity? In case this was not clear, currently Wayland users cannot resize any Electron-based windows. I don't think it can get much worse than that. In addition, even if this bug gets resolved quickly, it will take much more time to trickle into apps that bundle Electron, so it might also be worth opening separate issue for VS Code, Signal, etc. to let them know about it. This would probably also require a backport for v32, as these apps are not upgrading to the latest Electron version immediately.

**Comment by mhoffmeyerDC** [comment]
As a workaround, you can switch electron apps back to using x11/xwayland with the command flag:

```
--ozone-platform-hint=x11
```

**Comment by cbrnr** [comment]
This is not a viable option for me since I'm using fractional scaling on GNOME, which means that XWayland apps are blurry.

**Comment by charlag** [comment]
This affects our users also https://github.com/electron/electron/issues/44543

**Comment by beep-beep-beep-boop** [comment]
This bug makes most electron applications essentially unusable for me on wayland, which is the default display system of most linux distros nowadays. I'm not sure why it's not being treated with more severity. I can manually add the command-line argument to force things to use xwayland as a workaround, but that makes them blurry and doing it for every program is tedious, and I'm not sure that many users would be able to.

For example, one program I'd been using has been starting by default in this small size, and it is impossible to resize it:
![Image](https://github.com/user-attachments/assets/a28633b9-54b1-4ab8-9048-203f0797b966)

I've been having to find alternatives to electron software I'd been using due to this. I hope it's resolved quickly.

**Comment by morgan9e** [comment]
Any updates?

**Comment by morgan9e** [comment]
Any updates? I'm now experiencing it everywhere.. It's nearly unusable on Wayland

**Comment by EmptyVisual** [comment]
Hi @codebytere @deepak1556 

Apologies for the ping, and I hope I’m reaching out to the right people. From what I understand, you both might have the knowledge and expertise to help progress this issue (apologies if not).

I’ve bisected with electron fiddle and noticed that the resize issue seems to have originated between 32.0.0-alpha.5 (working resize) and 32.0.0-alpha.6 (broken resize). Interestingly, while alpha.6 appears to have resolved #42894, it also introduced the resizing issue we see in this issue, and I’m unsure of the direct relation or the specific commit involved. Replication can easily be done by running electron binary on a Wayland Session (in my case GNOME Wayland) with ./electron --ozone-platform-hint=auto

I tried my best to narrow this down further to specific commits but ran into challenges compiling from source. This has become a pretty big blocker for Wayland adoption, as many downstream projects are intentionally staying on a deprecated version of Electron and can’t move forward.

Any chance we could take another look at the priority of this issue when you have a moment? Thanks so much for your time ❤

**Comment by cbrnr** [comment]
> This has become a pretty big blocker for Wayland adoption, as many downstream projects are intentionally staying on a deprecated version of Electron and can’t move forward.

It's even worse, because if downstream projects upgrade to v32 or higher, their windows are not resizable. This is currently the case for Visual Studio Code, for example (I've linked the corresponding issue above).

**Comment by f0xb17** [comment]
The windows cannot be resized over the window edges, but Alt + F8 works fine with Gnome. It's crazy to do it like that, but if you currently need to resize the window, that's an option.

**Comment by aiddya** [comment]
The Windows Control Overlay PR is unlikely to be the cause. The resize issue can be bisected on the [32 branch](https://github.com/electron/electron/compare/v32.0.0-alpha.5...v32.0.0-alpha.6) and the [33 branch](https://github.com/electron/electron/compare/v33.0.0-nightly.20240701...v33.0.0-nightly.20240704) as well. The only common relevant change in both of those diffs is a Chromium version bump.

Unfortunately, the version bump has a [very large diff](https://chromium.googlesource.com/chromium/src/+log/127.0.6521.0..128.0.6571.0?pretty=fuller&n=10000) and it'll take some time to narrow down the issue. If anyone else wants to help, I'd be looking at desktop window tree host and frame view changes.

**Comment by f0xb17** [comment]
> The Windows Control Overlay PR is unlikely to be the cause. 

I don't think that has anything to do with it, either. 
The problem only affects Gnome, and Chromium-based browsers can be scaled without any problems. Yesterday, I also installed KDE to test and it works perfectly fine. 

I would argue that the issue has something to do with Client Side Decorations. Maybe you should take a look on that? Something changed in the past?

**Comment by darkblaze69** [comment]
this is not only in GNOME, also Weston is affected

**Comment by f0xb17** [comment]
> this is not only in GNOME, also Weston is affected

Just heard about it for the first time. Sorry for that, but found out that it seems like Weston is also using Client Side Decorations. Can you confirm that?

**Comment by cbrnr** [comment]
> I would argue that the issue has something to do with Client Side Decorations. Maybe you should take a look on that? Something changed in the past?

Maybe, but note that `electron31` works (i.e., can be resized), whereas `electron32` (and later) cannot be resized anymore. I've tested this on Arch using the packages `electron31` and `electron32` on GNOME/Wayland. So whatever it is, it did work with version 31 and stopped working after that.

**Comment by f0xb17** [comment]
Agree on that one. `electron31` is the last release that can be resized normally. With `electron32` it started. 

Perhaps we should look at what changes were made when we jumped from version 31 to version 32 and revert this step by step. 


Also keep in mind that there are some new flags for Wayland in Chromium that need to be activated separately:

- `Wayland per-window scaling`
- `Wayland text-input-v3`
- `Wayland UI scaling`
- `Wayland linux-drm-syncobj explicit sync`

**Comment by cbrnr** [comment]
My PC cannot handle building Electron from source (not enough SSD space, too slow), so unfortunately I cannot bisect. I'd be very happy if someone else could find the last working commit.

**Comment by EmptyVisual** [comment]
Alright, first of all—wow! Compiling Electron from source is definitely not for the faint of heart (or the impatient). It’s taken me the better part of two days to bisect as best I could, but I’m hopeful it’s been worth the effort! 😛 That said, I believe I’ve identified the issue commits. I’ve documented the earliest Electron release versions where we first observed these regressions, and further down, I’ve pinpointed the exact commit where the issues initially appeared. For reference, I’ve also linked the backports—just a heads-up, both issues seem to have originated from Chromium bumps.

To clarify further, the problem extends beyond the resize regression discussed in this issue. It’s likely that both this and the [incorrect CSD/inset](https://github.com/electron/electron/issues/42894#issuecomment-2351622791) issue share a common root cause.

Earliest we seen this in Releases:
- CSD Inset Regression first seen on [31.0.0-beta.1](https://github.com/electron/electron/compare/v31.0.0-alpha.5...v31.0.0-beta.1)
- Resize Regression first seen on [32.0.0-alpha.6](https://github.com/electron/electron/compare/v32.0.0-alpha.5...v32.0.0-alpha.6)


## Further drill down to the affected earliest commits:

### First Affected Commit (CSD Border Issue)

The CSD/inset issue started with the following commit:

`chore: bump chromium to 126.0.6445.0 (31-x-y)` (#42119)

The above included:

- **Upstream Chromium Change**: [5367794: wayland: Support window state and insets in configure-sync model](https://chromium-review.googlesource.com/c/chromium/src/+/5367794)
- **Electron Implementation**: [Address Linux NonClientFrameView Changes](https://github.com/electron/electron/commit/dff980c9c20da444fb853b6983eb75d48714b486)


Later, the Chromium developers reverted the upstream change, then re-landed it with code changes:

- **Revert Commit**: [5404542](https://chromium-review.googlesource.com/c/chromium/src/+/5404542)
- **Re-land Commit**: [5416939](https://chromium-review.googlesource.com/c/chromium/src/+/5416939/1..36)

You can compare the original patchset (1) with the re-landed patchset (36) on the re-land commit link to review the code changes.


### Second Affected Commit (This Issue)
**On the Electron end:**

The re-landed Chromium changes were pulled in and this is when the resize issue (this issue) first started to occur:

`chore: bump chromium to 128.0.6558.0 (32-x-y)` (#42746)

Testing these upstream Chromium changes with Chromium itself, I was not able to reproduce the issue, which leads me to believe it may likely be the Electron implementation commit [Address Linux NonClientFrameView Changes](https://github.com/electron/electron/commit/dff980c9c20da444fb853b6983eb75d48714b486).

The Electron changes may need to be reviewed again since upstream Chromium have re-landed theirs with new code changes. We may need to potentially look at the original Electron side implementation again as well and see why we ran into the CSD regression, but I suspect it may be the same root cause.

I don’t have deep knowledge of the codebase since my focus is primarily in QA, but I hope someone more familiar with the Electron codebase can make use of this information.

Not sure who to ping on this (just going on who was involved in the commits/reviews) since I know some of the team are on a quiet period, but hoping any of the talented engineers who have availability can at least offer some guidance/further comments here at all?
cc @codebytere @VerteDinde @jkleinsc @ckerr @deepak1556

**Comment by f0xb17** [comment]
First, thanks for your work, mate. 

In addition to your assumption:

https://github.com/electron/electron/issues/42894 can also be triggered with a Chromium-based browser and the [Rounded Window Corners Reborn Extension](https://github.com/flexagoon/rounded-window-corners)

If the window is minimized with the extension activated, exactly the same behavior occurs. 
I noticed this about 3–4 days ago with a stable version of Brave. At that time, `Wayland linux-drm-syncobj explicit sync` was not implemented in the stable branch. 

Since I am currently not working with Gnome, I can't check if the same behavior can still be triggered if `Wayland linux-drm-syncobj explicit sync` is activated. 

Hope this helps to narrow down the problem further.

**Comment by jonasmalacofilho** [comment]
> Testing these upstream Chromium changes with Chromium itself, I was not able to reproduce the issue, which leads me to believe it may likely be the Electron implementation commit [Address Linux NonClientFrameView Changes](https://github.com/electron/electron/commit/dff980c9c20da444fb853b6983eb75d48714b486).

I'm not sure if it's relevant, but that commit inverts a conditional: with it, `view->GetBorderDecorationInsets()` is used only if `native_window_view_->IsFullscreen()`, the opposite from before.

**Comment by f0xb17** [comment]
**_Update on Release 34:_**

Just installed Electron 34 (currently the one from Arch Linux repo). The problem is also included in this version.

**Comment by Smokolak** [comment]
Any progress on this issue? It affects many applications under Wayland.
I'm using Solus, running signal-desktop app, Visual Studio Code, etc. doesn't allow to resize the windows.

**Comment by krzemyczek** [comment]
> Any progress on this issue? 

As for me it is only annoying, it's more problematic for less tech people who don't know why...

**Comment by f0xb17** [comment]
> Any progress on this issue?

No new progress, still broken. Tested just a few seconds ago.
Maybe they will implement a fix, when Gnome 48 arrives in a few days.

**Comment by alyx161** [comment]
I don't see how gnome will fix a electron issue :(

**Comment by f0xb17** [comment]
> I don't see how gnome will fix a electron issue :(

Oh no = Maybe Electron Team will start working on a fix, when Gnome 48 is out. 
Only a few days left. 

Hope.

**Comment by cbrnr** [comment]
Love your optimism, but as far as I can tell, Electron devs have not even acknowledged this issue so far.

**Comment by f0xb17** [comment]
> Love your optimism, but as far as I can tell, Electron devs have not even acknowledged this issue so far.

Haha thanks. I hope the best, but expect the worst. 

This cannot remain forever, because it makes applications using Electron Framework unusable in some cases. 

Switching back to X11 or Xwayland can't be the solution.

Isn't Xwayland dropped with Gnome 48?

**Comment by TheNexter** [comment]
> Isn't Xwayland dropped with Gnome 48?

No, Xwayland is not dropped. There is only an option to build gnome without X11. But this is still a non default option for the moment.

But yes, please FIX this issue please.

**Comment by JacquesOfAllTrades** [comment]
> As for me it is only annoying, it's more problematic for less tech people who don't know why...

It's a big problem for screen-sharing, at least for me, because the window is too small to be able to read any text on the other person's screen.

There's no workaround, AFAICT. Popping the video into a new window or putting it into full-screen mode doesn't help. (The app thinks it's in full-screen, but it's actually still in the same, small window.) The stream is 1440p, so it's not that the source resolution is too low.

**Comment by codebytere** [comment]
I'll have a PR for this by the end of the week.

**Comment by f0xb17** [comment]
> I'll have a PR for this by the end of the week.


Thank you very much.
Looking forward to see a fix for that issue.

🙂

**Comment by trane77** [comment]
I'm not a electron developer, I just wanted to report FYI that this is happening to me on Fedora 42 (Gnome 48 on Wayland).

**Comment by alyx161** [comment]
@funder7 The issue is resolved since a while.
If you still experience this with some Electron applications, they are probably still using a older Electron version.

Probably the best idea to make sure you've installed the latest updates, and if you still have the issue create a bug report with the receptive app where you see this behavior.

**Comment by trane77** [comment]
Yep I've seen that, those apps are recent, so I'll try to report this to them, but seeing the problem on 2/2 apps, makes me think it's systematic.
